be/ut/dtm0_log_ut.c

```diff
@@ -605,13 +605,86 @@ static void m0_be_ut_dtm0_log_test(void)
 	M0_LEAVE();
 }
 
+
+static void m0_be_ut_dtm0_log_init_fini(void)
+{
+	struct m0_be_dtm0_log_iter iter;
+	struct m0_dtm0_clk_src     cs;
+	struct m0_be_dtm0_log     *log;
+	int rc;
+
+	m0_dtm0_clk_src_init(&cs, M0_DTM0_CS_PHYS);
+
+	rc = m0_be_dtm0_log_alloc(&log);
+	M0_UT_ASSERT(rc == 0);
+
+	rc = m0_be_dtm0_log_init(log, NULL, &cs, false);
+	M0_UT_ASSERT(rc == 0);
+
+	m0_be_dtm0_log_iter_init(&iter, log);
+	m0_be_dtm0_log_iter_fini(&iter);
+
+	m0_be_dtm0_log_fini(log);
+	m0_be_dtm0_log_free(&log);
+	m0_dtm0_clk_src_fini(&cs);
+}
+
+static void m0_be_ut_dtm0_log_next(void)
+{
+	struct m0_dtm0_tx_desc  txd = {};
+	struct m0_buf           buf = {};
+
+	struct m0_be_dtm0_log_iter iter;
+	struct m0_dtm0_log_rec	   out;
+	struct m0_dtm0_clk_src     cs;
+	struct m0_be_dtm0_log     *log;
+	int rc;
+
+	m0_dtm0_clk_src_init(&cs, M0_DTM0_CS_PHYS);
+
+	rc = m0_be_dtm0_log_alloc(&log);
+	M0_UT_ASSERT(rc == 0);
+
+	rc = m0_be_dtm0_log_init(log, NULL, &cs, false);
+	M0_UT_ASSERT(rc == 0);
+
+	rc = ut_dl_init(&txd, &buf, 42);
+	M0_UT_ASSERT(rc == 0);
+
+	p_state_set(&txd.dtd_ps.dtp_pa[0], M0_DTPS_PERSISTENT);
+	p_state_set(&txd.dtd_ps.dtp_pa[1], M0_DTPS_PERSISTENT);
+	p_state_set(&txd.dtd_ps.dtp_pa[2], M0_DTPS_PERSISTENT);
+	m0_mutex_lock(&log->dl_lock);
+	rc = m0_be_dtm0_log_update(log, NULL, &txd, &buf);
+	M0_UT_ASSERT(rc == 0);
+
+	m0_be_dtm0_log_iter_init(&iter, log);
+	rc = m0_be_dtm0_log_iter_next(&iter, &out);
+	M0_UT_ASSERT(rc == 1);
+	m0_dtm0_log_iter_rec_fini(&out);
```

Comment:
Can we check and verify the results in `out`?

Reply:
I agree.

Reply:
Fixed.

```
661         m0_be_dtm0_log_iter_init(&iter, log);
662         rc = m0_be_dtm0_log_iter_next(&iter, &out);
663         M0_UT_ASSERT(rc == 0);
664         M0_UT_ASSERT(ut_dl_verify_log_rec(&out, 42));
665         m0_dtm0_log_iter_rec_fini(&out);
```